[feat]: 카카오 OAuth 로그인 구현

TaskerJang opened the pull request.

Commit:
[build]: 프로젝트 기본 설정 파일 추가
- .gitignore 추가 (.env 파일 처리 관련)
- build.gradle 추가
- settings.gradle 변경 (대소문자 구분 이슈)

Commit:
[feat]: PasswordEncoder 빈 등록 추가
- BCryptPasswordEncoder 빈 설정 구현

Commit:
[feat]: 카카오 OAuth 로그인 구현
- User 엔티티에 Builder 패턴 적용하여 생성자 추가
- UserController에 카카오 로그인 및 현재 유저 조회 API 추가
- UserRepository에 이메일 기반 조회 메소드 추가
- UserService에 카카오 로그인 로직 구현

duhyeon99 reviewed (commented):
고생하셨습니다

duhyeon99 reviewed (approved):
고생하셨습니다

hwan2-99 commented on src/main/java/com/mindgoal/config/SecurityConfig.java:
```diff
@@ -21,6 +23,10 @@
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
     public static final String API_V_1 = "/"; //"/api/v1/";
```
해당부분 주석되있는 부분으로 수정하셔서 올려주시기로 했던것 같습니다!!

TaskerJang replied:
제가 착각한게 주석으로 수정안해도 되더라구요! 

 public static final String API_V_1 = "/";
 
 이게 맞았습니다 ㅎ.ㅎ

hwan2-99 commented on src/main/java/com/mindgoal/domain/user/entity/User.java:
```diff
@@ -39,6 +35,16 @@ public class User extends BaseEntity {
 
     @Column(name = "IS_AGREE_POLICY")
     private Boolean isAgreePolicy;
+    @Builder
```
빌더패턴 적용  :+1:

hwan2-99 commented on src/main/java/com/mindgoal/domain/user/service/UserService.java:
```diff
@@ -1 +1,95 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.reactive.function.client.WebClient;
+import java.util.UUID;
```
사소한거지만 import 문과 어노테이션문 간의 공백이 추가되면 좋을것 같습니다 :)

hwan2-99 reviewed (approved):
고생하셨습니다!

Commit:
[style/feat]: 코드 스타일 개선 및 로그아웃 기능 구현
- import문과 어노테이션 사이 공백 추가
- UserController에 로그아웃 엔드포인트 구현
- SecurityContext 초기화 로직 추가
